### PR TITLE
SCA with Php Inspections (EA Extended)

### PR DESCRIPTION
**Pull request by kalessil**
Minor code tweaks: automatic fixes by [Php Inspections (EA Extended)](https://plugins.jetbrains.com/plugin/7622-php-inspections-ea-extended-) (PhpStorm plugin).

- PhpStorm provides free [licenses for OSS projects](https://www.jetbrains.com/buy/opensource/) (and the analyzer is for the IDE only);
- [Getting started](https://github.com/kalessil/phpinspectionsea/blob/master/docs/getting-started.md) with Php Inspections (EA Extended)

### CODE REVIEW

**Comment by andrerom** [comment]
Nice, we're about to branch off 6.12, but would be nice to see this go in after if reviewers agree :)

File: eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php [hunk 1]
```diff
@@ -740,7 +740,7 @@ protected function getRealAliasId($locationId)
         $data = $this->gateway->loadAutogeneratedEntry('eznode:' . $locationId);
 
         // Root entries (URL wise) can return 0 as the returned value is used as parent (parent is 0 for root entries)
-        if (empty($data) || $data['id'] != 0 && $data['parent'] == 0 && strlen($data['text']) == 0) {
+        if (empty($data) || ($data['id'] != 0 && $data['parent'] == 0 && strlen($data['text']) == 0)) {
```
Comment (glye) [inline eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php]: This looks like a bug fix, not just improvement. Seems sane, but I hope you've checked that.

Comment (kalessil) [reply eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php]: Not really, rather code style - PHP interprets the code fragment this way.

Comment (alongosz) [reply eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php]: There's a flame war going on for ages about using brackets for expressions that are evaluated due to precedence. From my side +0.5 on this change ;)

Comment (glye) [reply eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php]: I'll certainly take logical grouping over relying on arbitrary precedence rules, any day. +1. 🔥

File: eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php [hunk 2]
```diff
@@ -140,7 +140,7 @@ public function index(FullTextData $fullTextData)
                 ++$wordCount;
                 // if we have "www." before word than
                 // treat it as url and add additional entry to the index
-                if (mb_substr(mb_strtolower($word), 0, 4) === 'www.') {
+                if (mb_strtolower(mb_substr($word, 0, 4)) === 'www.') {
```
Comment (alongosz) [inline eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php]: That's a very nice catch :)

Comment (kalessil) [reply eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php]: Indeed =)

Comment (kalessil) [reply eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php]: BTW, it's one of the inspections in Php Inspections (EA Extended). What's cool is that all changes made automatically (quick-fixes).

**Comment by alongosz** [comment]
@kalessil it just occurred to me - could you add in the PR description something about the tool you're using and how to perform these automatic fixes in the future?

**Comment by kalessil** [comment]
@alongosz : done, all is in the PR description

**Comment by glye** [comment]
@andrerom Too late for 6.12 I guess, but should be merged anyway. Should we take it back to 6.7?

**Comment by andrerom** [comment]
@glye we can do that, to avoid conflicts on any of this.

**Comment by andrerom** [comment]
Thanks @kalessil 👍

**Pull request merged**